### PR TITLE
Simplify preflight validation check

### PR DESCRIPTION
**Pull request by jsonar-cpapke**
Related to https://onejira.imperva.com/browse/SD-1064
Reduce preflight validation stack to only 3 functions

### CODE REVIEW

**Comment by lindanasredin** [comment]
Hi Chris, I reviewed this with Sivan. 
1. Regarding replacing some function arguments with 'args', there are a few issues with this change. When we wrote the code, we planned to extract some of the functions to another file to reduce the size of the main.py file, the other file won't have args in its context and needs to receive explicit arguments. In addition, using args in the inner functions makes the code less clear - it is not clear which arguments the functions uses and which it doesn't.
2. Regarding renaming extended_node to node: Node is the structure we receive as input from the user. Extended node has this node and other fields (see create_extended_node function). These are two different terms and not interchangeable both in function names and argument/field names.
3. Reducing the function stack means that functions are longer. This is a matter of style.
4. The main issue - the change in function run_preflight_validations (see comment in the code) broke the automation when stop_on_failure = false. In this case, if one of the nodes' validations failed, it won't continue to the next nodes.
In general, please make sure the automation passes before assigning a reviewer, this is our way of knowing the code is ready for review. Sometimes one creates a PR before wanting a review.
5. Passing the POSTFLIGHT_VALIDATIONS_SCRIPT_NAME argument to inner functions is a good idea. Please also change it for UPGRADE_SCRIPT_NAME, POSTFLIGHT_VALIDATIONS_SCRIPT_NAME and CLEAN_OLD_DEPLOYMENTS_SCRIPT_NAME to make it consistent.

File: modules/aws/sonar-upgrader/python_upgrader/upgrade/main.py [hunk 1]
```diff
@@ -415,79 +417,63 @@ def collect_python_location(extended_node, stop_on_failure, upgrade_status_servi
...
-        success_or_skip = maybe_run_preflight_validations_for_extended_node(extended_node, target_version,
-                                                                            script_file_name, stop_on_failure,
-                                                                            upgrade_status_service)
-        all_success_or_skip = all_success_or_skip and success_or_skip
-    return all_success_or_skip
-
+    successful = True
+    for node in chain(agentless_gw_extended_node_dict.values(), dsf_hub_extended_node_dict.values()):
+        if upgrade_status_service.should_run_preflight_validations(node.get('dsf_node_id')):
+            successful = successful and run_preflight_validations_for_node(args, node, upgrade_status_service)
```
Comment (lindanasredin) [inline modules/aws/sonar-upgrader/python_upgrader/upgrade/main.py]: 4. The main issue - the change in function run_preflight_validations (see comment in the code) broke the automation when stop_on_failure = false. In this case, if one of the nodes' validations failed, it won't continue to the next nodes.

**Comment by jsonar-cpapke** [comment]
The only function that I see that would be longer than before is `run_preflight_validations_script`, the other have actually reduce their size.

**Comment by jsonar-cpapke** [comment]
As for args, I don't see the point of passing around multiple individual args, when the value of them is explicitly from the user. Grabbing the values from the argparse results makes the number of arguments per function _significantly_ less, and makes it obvious it isn't a generated value.

File: modules/aws/sonar-upgrader/python_upgrader/upgrade/main.py (hunk 1 above)
Comment (jsonar-cpapke) [reply modules/aws/sonar-upgrader/python_upgrader/upgrade/main.py]: Short circuit issue has been fixed.

**Review by sivan-hajbi-imperva — APPROVED** [review]

**Pull request merged**